client/src/components/ResourceForm.js

```diff
@@ -70,10 +70,25 @@ const ResourceForm = (props: FormProps) => {
       if (id) {
         const resource = await getResourceByID(id);
         if (resource) {
-          const { name, ...result } = resource.result;
-          setFieldsValue({ resourceName: name, ...result });
-          setCategory(result.category);
-          setSubcategory(result.subcategory);
+          const { name, category, subcategory, ...result } = resource.result;
+
+          // generate initial category values
+          const categoryInitialValues = [];
+          for (let i = 0; i < category.length; ) {
+            categoryInitialValues.push(
+              category[i] + CAT_SUB_SPLITTER + subcategory[i],
+            );
+
+            i += 1;
```

Comment:
Why the split?

Reply:
That way we can denote which subcategory belongs to what category.